virtcontainers/cgroups.go

```diff
@@ -190,6 +193,29 @@ func renameCgroupPath(path string) (string, error) {
 
 }
 
+// validCgroupPath returns a valid cgroup path.
+// see https://github.com/opencontainers/runtime-spec/blob/master/config-linux.md#cgroups-path
+func validCgroupPath(path string, systemdCgroup bool) (string, error) {
+	if isSystemdCgroup(path) {
+		return path, nil
+	}
+
+	if systemdCgroup {
+		return "", fmt.Errorf("malformed systemd path '%v': expected to be of form 'slice:prefix:name'", path)
+	}
+
+	// In the case of an absolute path (starting with /), the runtime MUST
+	// take the path to be relative to the cgroups mount point.
+	if filepath.IsAbs(path) {
+		return renameCgroupPath(filepath.Clean(path))
+	}
+
+	// In the case of a relative path (not starting with /), the runtime MAY
+	// interpret the path relative to a runtime-determined location in the cgroups hierarchy.
+	// clean up path and return a new path relative to defaultCgroupPath
+	return renameCgroupPath(filepath.Join(defaultCgroupPath, filepath.Clean("/"+path)))
```

Comment:
AFAIU, we shouldn't really ever hit this point?  When utilizing cgroupfs, it is always an absolute path, and in the case of systemd-cgroups, it is usually relative?  Correct?

If so, can we state that this is an unexpected scenario

Reply:
don't know if this is expected or not, but it's documented in the OCI specification https://github.com/opencontainers/runtime-spec/blob/master/config-linux.md#control-groups

>  In the case of an absolute path (starting with /), the runtime MUST take the path to be relative to the cgroups mount point.
    In the case of a relative path (not starting with /), the runtime MAY interpret the path relative to a runtime-determined location in the cgroups hierarchy.